tst/com/amazon/ion/benchmark/DataGeneratorTest.java

```diff
@@ -0,0 +1,209 @@
+package com.amazon.ion.benchmark;
+
+import com.amazon.ion.IonReader;
+import com.amazon.ion.IonType;
+import com.amazon.ion.Timestamp;
+import com.amazon.ion.system.IonReaderBuilder;
+import com.amazon.ion.util.IonStreamUtils;
+import org.junit.After;
+import org.junit.Test;
+
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+public class DataGeneratorTest {
+
+    public static String outputFile;
+
+    /**
+     * Construct IonReader for current output file in order to finish the following test process
+     * @param optionsMap is the hash map which generated by the command line parser which match the option name and its value appropriately.
+     * @return constructed IonReader
+     * @throws Exception if errors occur during executing data generator process.
+     */
+    public static IonReader constructReader(Map<String, Object> optionsMap) throws Exception {
+        outputFile = optionsMap.get("<output_file>").toString();
+        GeneratorOptions.executeGenerator(optionsMap);
+        return IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(outputFile)));
+    }
+
+    /**
+     * Assert generated Ion data is the same type as expected.
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedType() throws Exception {
+        Map<String, Object> optionsMap = Main.parseArguments("generate", "-S", "500", "-T", "decimal", "test1.10n");
```

Comment:
> For the tests, since brevity isn't a concern, I recommend using the long option names. I think that will make it easier to see what the test is testing without having to remember what the short names stand for.

All short names have been changed to the original long option names.